packages/client/src/app/pages/portfolio/blogs/blogs.component.ts

```diff
@@ -0,0 +1,162 @@
+import { AuthService } from '../../../services/auth';
+import { Constants, Title } from '../../../shared';
+import { PortBlogs } from '../../../models/site';
+import { MatDialog } from '@angular/material/dialog';
+import { CreateBlogComponent, EditBlogComponent, PreviewBlogComponent } from '../../../components/modals/blogs';
+import { BlogsService } from '../../../services/content';
+
+@Component({
+    selector: 'port-blogs',
+    templateUrl: './blogs.component.html',
+    styleUrls: ['./blogs.component.less']
+})
+export class BlogsComponent implements OnInit {
+    currentUser: FrontendUser;
+    portUser: FrontendUser;
+    blogsData: PaginateResult<Blog>
+    userBlogsData: PaginateResult<Blog>
+
+    pageNum = 1;
+    listView = false;
+
+    searchBlogs = new FormGroup({
+        query: new FormControl('')
+    });
+
+    constructor(private route: ActivatedRoute, private router: Router, private authService: AuthService, private dialog: MatDialog, private blogsService: BlogsService) {
+        this.authService.currUser.subscribe(x => {
+            this.currentUser = x;
+        });
+    }
+
+    ngOnInit(): void {
+        this.portUser = this.route.parent.snapshot.data.portData as FrontendUser;
+        Title.setThreePartTitle(this.portUser.username, Constants.BLOGS);
+
+        this.route.data.subscribe(data => {
+            const feedData = data.feedData as PortBlogs;
+            this.blogsData = feedData.blogs;
+            this.userBlogsData = feedData.userBlogs;
+        });
+    }
+
+    /**
+     * Handles page changing
+     * 
+     * @param event The new page
+     */
+    onPageChange(event: number) {
+        this.router.navigate([], {relativeTo: this.route, queryParams: {page: event}, queryParamsHandling: 'merge'});
+        this.pageNum = event;
+    }
+
+    /**
+     * Checks to see if the currently logged in user is the same as the user who owns this portfolio.
+     */
+    currentUserIsSame() {
+        if (this.currentUser) {
+            if (this.portUser) {
+                if (this.currentUser._id === this.portUser._id) {
+                    return true;
+                } else {
+                return false;
+                }
+            }
+        }
+    }
+
+    /**
+     * Switches the view
+     */
+    switchView() {
+        if (this.listView === true) {
+            this.listView = false;
+            this.router.navigate([], {relativeTo: this.route, queryParams: {page: 1}, queryParamsHandling: 'merge'});
+        } else {
+            this.listView = true;
+            this.router.navigate([], {relativeTo: this.route, queryParams: {page: 1}, queryParamsHandling: 'merge'});
+        }
+    }
+
+  /**
+   * Opens the new blog form.
+   */
+  openNewBlogForm() {
+    const createBlogRef = this.dialog.open(CreateBlogComponent);
+    createBlogRef.afterClosed().subscribe(() => {
+      this.router.navigate([], {relativeTo: this.route, queryParams: {page: this.pageNum}, queryParamsHandling: 'merge'});
```

Comment:
Does this navigate unconditionally, even if the user cancelled the dialogue, or didn't make a choice?

Reply:
yeah, it always navigates